plugins/google/pkg/app/fetch.go

```diff
@@ -0,0 +1,106 @@
+package app
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/alecthomas/kong"
+	"github.com/aserto-dev/ds-load/plugins/google/pkg/googleclient"
+	"github.com/aserto-dev/ds-load/sdk/plugin"
+)
+
+type FetchCmd struct {
+	ClientID     string `short:"i" help:"Google Client ID" env:"GOOGLE_CLIENT_ID" required:""`
+	ClientSecret string `short:"s" help:"Google Client Secret" env:"GOOGLE_CLIENT_SECRET" required:""`
+	RefreshToken string `short:"r" help:"Google Refresh Token" env:"GOOGLE_REFRESH_TOKEN" required:""`
+	Groups       bool   `short:"g" help:"Retrieve Google groups" env:"GOOGLE_GROUPS" default:"false" negatable:""`
+}
+
+func (cmd *FetchCmd) Run(ctx *kong.Context) error {
+	googleClient, err := createGoogleClient(cmd.ClientID, cmd.ClientSecret, cmd.RefreshToken)
+	if err != nil {
+		return err
+	}
+
+	results := make(chan map[string]interface{}, 1)
+	errors := make(chan error, 1)
+	go func() {
+		Fetch(googleClient, cmd.Groups, results, errors)
+		close(results)
+		close(errors)
+	}()
+	return plugin.NewDSPlugin().WriteFetchOutput(results, errors, false)
+}
+
+func Fetch(googleClient *googleclient.GoogleClient, fetchGroups bool, results chan map[string]interface{}, errors chan error) {
+	users, err := googleClient.ListUsers()
+	if err != nil {
+		errors <- err
```

Comment:
these errors are being written to stderr, but the plugin will still exit with 0. we should set the exit code like so: https://github.com/aserto-dev/ds-load/blob/main/plugins/auth0/pkg/app/fetch.go#L90

Reply:
I added calls to all the places where we produce an error.